buildmodel/buildmodel_test.go

```diff
@@ -5,14 +5,24 @@ package buildmodel
 
 import (
 	"errors"
-	"reflect"
+	"flag"
+	"os"
+	"path/filepath"
 	"testing"
 
+	"github.com/go-test/deep"
 	"github.com/microsoft/go-infra/buildmodel/buildassets"
 	"github.com/microsoft/go-infra/buildmodel/dockermanifest"
 	"github.com/microsoft/go-infra/buildmodel/dockerversions"
 )
 
+var update = flag.Bool("update", false, "Update the golden files instead of failing.")
+
+func TestMain(m *testing.M) {
```

Comment:
You don't need to call `flag.Parse()` in test code, it's done implicitly, and it's only necessary if `TestMain` depends on a flag, which is not the case:
https://pkg.go.dev/testing#hdr-Main